lib/dependencies.sh

```diff
@@ -3,7 +3,7 @@ run_if_present() {
   local has_script=$(read_json "$APP_DIR/package.json" ".scripts[\"$script_name\"]")
   if [ -n "$has_script" ]; then
     echo "Running $script_name"
-    npm run "$script_name" --if-present
+    yarn "$script_name"
```

Comment:
This version of heroku buildpack is quite old and doesn't install yarn. Only running this won't be enough 😃 
A couple ways of fixing this:
1. Sync the latest version of the buildpack with this repo, and then make changes to it so that it starts considering `client` as the root folder (just like we have edited in this version, but things probably changed a lot in latest one).
2. Go to the binaries file and hook yarn installation somewhere in [install_nodejs](https://github.com/moteef/heroku-buildpack-nodejs/blob/d6dc843148f6d1d345f7562f0145be50467828c6/lib/binaries.sh#L10)  or [install_npm](https://github.com/moteef/heroku-buildpack-nodejs/blob/d6dc843148f6d1d345f7562f0145be50467828c6/lib/binaries.sh#L48).
You can do it from [Debian package repository](https://yarnpkg.com/lang/en/docs/install/#linux-tab) or with [npm install](https://yarnpkg.com/lang/en/docs/install/#alternatives-tab), but they recommend installing from the package manager.

I would prefer option 1, because they already have everything figured out (identify if it's yarn from `yarn.lock` file, correctly cache yarn dependencies etc). Changing the code to use `client` as root may seem harder but I think it will be faster in the end :)

Reply:
And to test your changes to the buildpack you need to update `.buildpacks` file in moteef repo to use this branch (`https://github.com/moteef/heroku-buildpack-nodejs.git#feature/use-yarn`). Then on each push you can go to the review app [build logs](https://dashboard.heroku.com/apps/moteefe-staging1-pr-3729/activity) and see if everything is working. Please ensure that cache is correctly working 😃